Add STF 1.3 to 1.4 upgrade guide (#373)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_removing-service-telemetry-framework-1-3-operators.adoc

[id="removing-service-telemetry-framework-1-3-operators_{context}"]
= Removing {Project} 1.3 Operators

[role="_abstract"]
Remove the Smart Gateway Operator from {ProjectShort} 1.3.

.Procedure

> . Retrieve the `Subscription` name of the Smart Gateway Operator. Replace `service-telemetry` in the selector with the namespace hosting your {ProjectShort} instance if it is different from the default. Verify that only one subscription is returned.

Reviewer: ```suggestion
. Retrieve the `Subscription` name of the Smart Gateway Operator. Replace `service-telemetry` in the selector with the namespace that hosts your {ProjectShort} instance if it is different from the default namespace. Verify that only one subscription is returned.
```

+
[source,bash,options="nowrap"]
----
$ oc get sub --selector=operators.coreos.com/smart-gateway-operator.service-telemetry

NAME                                                                       PACKAGE                  SOURCE             CHANNEL
smart-gateway-operator-stable-1.3-redhat-operators-openshift-marketplace   smart-gateway-operator   redhat-operators   stable-1.3
----

. Delete the Smart Gateway Operator subscription:
+
[source,bash,options="nowrap"]
----
$ oc delete sub --selector=operators.coreos.com/smart-gateway-operator.service-telemetry

subscription.operators.coreos.com "smart-gateway-operator-stable-1.3-redhat-operators-openshift-marketplace" deleted
----

> . Retrieve the Smart Gateway Operator ClusterServiceVersion. Verify that only one ClusterServiceVersion is returned.

Reviewer: ```suggestion
. Retrieve the Smart Gateway Operator ClusterServiceVersion and verify that only one ClusterServiceVersion is returned:
```

----
$ oc get csv --selector=operators.coreos.com/smart-gateway-operator.service-telemetry

NAME                                     DISPLAY                  VERSION          REPLACES   PHASE
smart-gateway-operator.v3.0.1635451893   Smart Gateway Operator   3.0.1635451893              Succeeded
----

. Delete the Smart Gateway Operator ClusterServiceVersion:
+
[source,bash,options="nowrap"]
----
$ oc delete csv --selector=operators.coreos.com/smart-gateway-operator.service-telemetry

clusterserviceversion.operators.coreos.com "smart-gateway-operator.v3.0.1635451893" deleted
----

. Delete the SmartGateway Custom Resource Definition:
+
[source,bash,options="nowrap"]
----
$ oc delete crd smartgateways.smartgateway.infra.watch

customresourcedefinition.apiextensions.k8s.io "smartgateways.smartgateway.infra.watch" deleted
----

. Patch the Service Telemetry Operator Subscription to use the stable-1.4 channel:
+
[source,bash,options="nowrap"]
----
$ oc patch $(oc get sub --selector=operators.coreos.com/service-telemetry-operator.service-telemetry -oname) --patch $'spec:\n  channel: stable-1.4' --type=merge

subscription.operators.coreos.com/service-telemetry-operator patched
----

> . Monitor the output of the `oc get csv` command until the Smart Gateway Operator is installed and Service Telemetry Operator is `Pending` for version 1.3 and 1.4. Note the name of the service-telemetry-operator.v1.3 CSV in phase Replacing.

Reviewer: ```suggestion
. Monitor the output of the `oc get csv` command until the Smart Gateway Operator is installed and Service Telemetry Operator is `Pending` for version 1.3 and 1.4. Note the name of the service-telemetry-operator.v1.3 CSV in phase Replacing.
```

Should "Replacing" be "Replaces"?

Author: This is the name of the phase it is in. We can wrap it backticks or add emphasis if you wish. The output is the literal word "Replacing".

amq7-cert-manager.v1.0.3                     Red Hat Integration - AMQ Certificate Manager   1.0.3            amq7-cert-manager.v1.0.2                     Succeeded
amq7-interconnect-operator.v1.10.5           Red Hat Integration - AMQ Interconnect          1.10.5           amq7-interconnect-operator.v1.10.4           Succeeded
elasticsearch-eck-operator-certified.1.9.1   Elasticsearch (ECK) Operator                    1.9.1                                                         Succeeded
prometheusoperator.0.47.0                    Prometheus Operator                             0.47.0           prometheusoperator.0.37.0                    Succeeded
service-telemetry-operator.v1.3.1635451892   Service Telemetry Operator                      1.3.1635451892                                                Replacing
service-telemetry-operator.v1.4.1641504218   Service Telemetry Operator                      1.4.1641504218   service-telemetry-operator.v1.3.1635451892   Pending
smart-gateway-operator.v4.0.1641504216       Smart Gateway Operator                          4.0.1641504216                                                Succeeded
----

> . Delete the Service Telemetry Operator v1.3 ClusterServiceVersion that was noted in the output of `oc get csv`:

Reviewer: ```suggestion
. Delete the Service Telemetry Operator v1.3 ClusterServiceVersion that you noted in the output of the `oc get csv` command:
```
